adam-cli/pom.xml

```diff
@@ -14,22 +14,35 @@
   <build>
     <plugins>
       <plugin>
-        <groupId>org.codehaus.mojo</groupId>
-        <artifactId>appassembler-maven-plugin</artifactId>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-shade-plugin</artifactId>
         <configuration>
-          <programs>
-            <program>
-              <mainClass>org.bdgenomics.adam.cli.ADAMMain</mainClass>
-              <id>adam</id>
-            </program>
-          </programs>
+          <createDependencyReducedPom>true</createDependencyReducedPom>
+          <filters>
+            <filter>
+              <artifact>*:*</artifact>
+              <excludes>
+                <exclude>META-INF/*.SF</exclude>
+                <exclude>META-INF/*.DSA</exclude>
+                <exclude>META-INF/*.RSA</exclude>
+              </excludes>
+            </filter>
+          </filters>
```

Comment:
This is definitely not ready to go yet.

It still needs
 - removal of ```compute-adam-classpath.sh``` and ```compute-adam-jars.sh```; I haven't done this yet because I'm not sure how ```adam-pyspark``` is supposed to work
 - a careful look at maven-shade-plugin excludes/transformer configuration as mentioned above
 - decision on whether to continue using cli module as the assembly artifact or create a separate assembly module as in the Spark build
 - cleanup of changes to ```adam-shell.sh``` and ```adam-submit.sh``` per review notes
 - test assembly artifact on YARN cluster, to confirm issues such as #663 don't resurface
 - test assembly artifact in build for downstream applications